src/server.jl

```diff
@@ -108,6 +105,77 @@ function evaluate!(
     end
 end
 
+function _extract_relevant_frontmatter(file_frontmatter::Dict, options::Dict)
+    file_frontmatter = _recursive_merge(default_frontmatter(), file_frontmatter)
+
+    fig_width_default = get(file_frontmatter, "fig-width", nothing)
+    fig_height_default = get(file_frontmatter, "fig-height", nothing)
+    fig_format_default = get(file_frontmatter, "fig-format", nothing)
+    fig_dpi_default = get(file_frontmatter, "fig-dpi", nothing)
+
+    pandoc_to_default = nothing
+
+    julia_default = get(file_frontmatter, "julia", nothing)
+
+    if isempty(options)
+        return _frontmatter_template(;
+            fig_width = fig_width_default,
+            fig_height = fig_height_default,
+            fig_format = fig_format_default,
+            fig_dpi = fig_dpi_default,
+            pandoc_to = pandoc_to_default,
+            julia = julia_default,
+        )
+    else
+        D = Dict{String,Any}
+
+        format = get(D, options, "format")
+        execute = get(D, format, "execute")
+        fig_width = get(execute, "fig-width", fig_width_default)
+        fig_height = get(execute, "fig-height", fig_height_default)
+        fig_format = get(execute, "fig-format", fig_format_default)
+        fig_dpi = get(execute, "fig-dpi", fig_dpi_default)
+
+        pandoc = get(D, format, "pandoc")
+        pandoc_to = get(pandoc, "to", pandoc_to_default)
+
+        metadata = get(D, format, "metadata")
+        julia = get(metadata, "julia", julia_default)
+
+        return _frontmatter_template(;
+            fig_width,
+            fig_height,
+            fig_format,
+            fig_dpi,
+            pandoc_to,
+            julia,
+        )
+    end
+end
+
+function _frontmatter_template(;
+    fig_width,
+    fig_height,
+    fig_format,
+    fig_dpi,
+    pandoc_to,
+    julia,
+)
+    D = Dict{String,Any}
+    return D(
+        "format" => D(
+            "execute" => D(
+                "fig-width" => fig_width,
+                "fig-height" => fig_height,
+                "fig-format" => fig_format,
+                "fig-dpi" => fig_dpi,
+            ),
+            "pandoc" => D("to" => pandoc_to),
+            "metadata" => D("julia" => julia),
+        ),
+    )
```

Comment:
I decided just to trim down what gets passed to the workers rather than send that entire object. If we find we need any additional specific keys then we can add them on an as-needed basis rather than taking the kitchen sink approach.

Reply:
sounds all right to me